Unable to transform class related to record.

Consider the following use case:
```java
package com.example;
public record ExampleRecord(String name) {}
```
```java
package com.example.mixin;

import com.example.ExampleRecord;
import net.minecraft.client.main.Main;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(Main.class)
public class MixinMain {
    @Inject(method = "main", at = @At("HEAD"))
    private static void inject$main(String[] args, CallbackInfo ci) {
        try {
            ExampleRecord record = new ExampleRecord("");
        } catch (Throwable t) {}
    }
}
```

The above code will throw the following exception:
```
Exception in thread "main" java.lang.RuntimeException: java.lang.UnsupportedOperationException: Record requires ASM8
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:39)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.run(Launcher.java:106)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.Launcher.main(Launcher.java:77)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@0.1.17/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:90)
Caused by: java.lang.UnsupportedOperationException: Record requires ASM8
	at org.objectweb.asm@9.1.0/org.objectweb.asm.ClassVisitor.visitRecordComponent(ClassVisitor.java:305)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.ClassReader.readRecordComponent(ClassReader.java:930)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.ClassReader.accept(ClassReader.java:708)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.ClassReader.accept(ClassReader.java:401)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.TransformerClassWriter.computeHierarchyFromFile(TransformerClassWriter.java:151)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:112)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.TransformerClassWriter.getSupers(TransformerClassWriter.java:83)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.TransformerClassWriter.getCommonSuperClass(TransformerClassWriter.java:63)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.SymbolTable.addMergedType(SymbolTable.java:1202)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.Frame.merge(Frame.java:1299)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.Frame.merge(Frame.java:1197)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.MethodWriter.computeAllFrames(MethodWriter.java:1610)
	at org.objectweb.asm@9.1.0/org.objectweb.asm.MethodWriter.visitMaxs(MethodWriter.java:1546)
	at org.objectweb.asm.tree@9.1.0/org.objectweb.asm.tree.MethodNode.accept(MethodNode.java:769)
	at org.objectweb.asm.tree@9.1.0/org.objectweb.asm.tree.MethodNode.accept(MethodNode.java:649)
	at org.objectweb.asm.tree@9.1.0/org.objectweb.asm.tree.ClassNode.accept(ClassNode.java:452)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:133)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50)
	at cpw.mods.securejarhandler@0.9.46/cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:110)
	at cpw.mods.securejarhandler@0.9.46/cpw.mods.cl.ModuleClassLoader.lambda$findClass$16(ModuleClassLoader.java:213)
	at cpw.mods.securejarhandler@0.9.46/cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:223)
	at cpw.mods.securejarhandler@0.9.46/cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:213)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:631)
	at java.base/java.lang.Class.forName(Class.java:543)
	at MC-BOOTSTRAP/fmlloader@1.17.1-37.0.61/net.minecraftforge.fml.loading.targets.ForgeClientUserdevLaunchHandler.lambda$launchService$0(ForgeClientUserdevLaunchHandler.java:38)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@9.0.7/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37)
	... 7 more
```

This is caused by `SuperCollectingVisitor` using ASM7 which does not support record:
https://github.com/cpw/modlauncher/blob/8569cdfcbc41cf97d7fcf94e34789072af64c0a8/src/main/java/cpw/mods/modlauncher/TransformerClassWriter.java#L168-L170

--- a/src/main/java/cpw/mods/modlauncher/ClassTransformer.java
+++ b/src/main/java/cpw/mods/modlauncher/ClassTransformer.java
@@ -68,7 +68,7 @@ public class ClassTransformer {
             return inputClass;
         }
 
-        ClassNode clazz = new ClassNode(Opcodes.ASM7);
+        ClassNode clazz = new ClassNode(Opcodes.ASM9);
         Supplier<byte[]> digest;
         boolean empty;
         if (inputClass.length > 0) {

--- a/src/main/java/cpw/mods/modlauncher/PredicateVisitor.java
+++ b/src/main/java/cpw/mods/modlauncher/PredicateVisitor.java
@@ -25,23 +25,25 @@ import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 
 public class PredicateVisitor extends ClassVisitor {
+    private static final int ASM_API = Opcodes.ASM9;
+    
     private ITransformerVotingContext.MethodPredicate methodPredicate;
     private ITransformerVotingContext.FieldPredicate fieldPredicate;
     private ITransformerVotingContext.ClassPredicate classPredicate;
     private boolean result;
 
     PredicateVisitor(final ITransformerVotingContext.FieldPredicate fieldPredicate) {
-        super(Opcodes.ASM7);
+        super(ASM_API);
         this.fieldPredicate = fieldPredicate;
     }
 
     PredicateVisitor(final ITransformerVotingContext.MethodPredicate methodPredicate) {
-        super(Opcodes.ASM7);
+        super(ASM_API);
         this.methodPredicate = methodPredicate;
     }
 
     PredicateVisitor(final ITransformerVotingContext.ClassPredicate classPredicate) {
-        super(Opcodes.ASM7);
+        super(ASM_API);
         this.classPredicate = classPredicate;
     }
 

--- a/src/main/java/cpw/mods/modlauncher/TransformerClassWriter.java
+++ b/src/main/java/cpw/mods/modlauncher/TransformerClassWriter.java
@@ -166,7 +166,7 @@ class TransformerClassWriter extends ClassWriter {
     private class SuperCollectingVisitor extends ClassVisitor {
 
         public SuperCollectingVisitor() {
-            super(Opcodes.ASM7);
+            super(Opcodes.ASM9);
         }
 
         @Override

